modules/fedimint-ln-server/src/lib.rs

```diff
@@ -885,11 +919,60 @@ impl ServerModule for Lightning {
 }
 
 impl Lightning {
-    pub fn new(cfg: LightningConfig, task_group: &mut TaskGroup) -> anyhow::Result<Self> {
+    pub async fn new(
+        cfg: LightningConfig,
+        db: Database,
+        task_group: &mut TaskGroup,
+    ) -> anyhow::Result<Self> {
         let btc_rpc = create_bitcoind(&cfg.local.bitcoin_rpc, task_group.make_handle())?;
+        task_group
+            .spawn("compact audit keys", |handle| async move {
+                Self::compact_audit_keys_task(db, &handle).await;
+            })
+            .await;
         Ok(Lightning { cfg, btc_rpc })
     }
 
+    async fn compact_audit_keys_task(db: Database, tg_handle: &TaskHandle) {
+        let interval = Duration::from_secs(60 * 5);
+        while !tg_handle.is_shutting_down() {
+            let mut dbtx = db.begin_transaction().await;
+            Self::compact_audit_keys(&mut dbtx).await;
+            if dbtx.commit_tx_result().await.is_err() {
+                warn!(
+                    "conflict while compacting audit keys, retrying in {} seconds",
+                    interval.as_secs()
+                );
+            }
+            sleep(interval).await;
+        }
+    }
+
+    /// Runs compaction for LightningAuditItemKeys
+    ///
+    /// When a contract reaches a terminal state, the associated amount will be
+    /// updated to 0. At this point, the contract no longer needs to be tracked
+    /// for auditing liabilities, so we can safely remove the audit key.
+    async fn compact_audit_keys(dbtx: &mut DatabaseTransaction<'_>) {
+        let _timing /* logs on drop */ = TimeReporter::new("compact_audit_keys");
+        let remove_audit_keys = dbtx
+            .find_by_prefix(&LightningAuditItemKeyPrefix)
+            .await
+            .filter_map(|(key, amount)| async move {
```

Comment:
nit: this could be just `filter(|(_, amount) amount == 0)`? It doesn't seem to transform what it's filtering.

Reply:
I wanted to return only the key instead of a tuple of `(key, amount)` so `filter_map` was used here to transform the return type. That said, it may be clearer to use only `filter` and handle the tuple downstream, but this logic has been removed with the latest update.